### DISCUSSION TITLE
Apa102 directly on the Pi

### DISCUSSION THREAD

**Discussion by dannyv1101** [awawa-dev/HyperHDR#574] [General]
Hi, I just saw this project and I like it very much. But I have some questions before I start with it. 
I am curently running an old version of hyperion and I want to update my system. 
I have apa102 attached on my tv. I just want to test hyperHDR before I remove everything. 
So my question: will it work with apa102 led stripes directly connected to the raspi (without the esp), like it is now with my hyperion build.

**Comment by awawa-dev** [comment]
Hi
If you don't change the hardware configuration but just change the application, I don't see why it won't work. It is generally not recommended to drive SPI LEDs (apa102) from Raspberry Pi without 3.3V to 5V level shifter, but since the current setup works for you, it will remain the same in this case. Having a backup of configuration is always a good idea if you plan to upgrade any application.

**Reply by dannyv1101** [reply]
Yes I will use a new SD Card and keep the old one as backup. Yes I have a level shifter in my setup.

I have one more question, is it possible to control hyperHDR with iobroker? There is a HyperionNG adapter, will it also work with hyperHDR?

**Reply by awawa-dev** [reply]
You can use JSON API commands to control HyperHDR e.g. using GET or POST requests. The wizard can help you to build commands.
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/99b628e9-a594-4da8-834d-dc902aaea0f9)
Since HyperHDR and Hyperion API are different I doubt if iobroker old plugins will be compatible.

**Comment by dannyv1101** [comment]
Oh that is a very nice feature. Thank you.

**Comment by dannyv1101** [comment]
Do I have to setup in HyperHDR if my LED Stripe is connected without an esp32 (directly to the Pi) or does it work automatically?

**Reply by awawa-dev** [reply]
For SPI driver must select proper path (SPI 0.0 is default) and enable it, details can be found on the logs subpage.

**Comment by dannyv1101** [comment]
Ok I yesterday have changed to HyperHDR, and everthing works great. The old remote commands for power on and power also still work, but I think I will change them to the new one. So the next step is to connect the esp32 with Hyperserial.

Thank for this great work!!!

**Comment by PaulePils** [comment]
Sorry to use this old thread but I am currently trying the same (migrate from Hyperion to HyperHDR) and got some errors.
I have a Raspi 4 with libreElec 9.2.6 with an APA102 (unfortunatly the cheap version, just found out after reading your tutorial) and used the internal grabber. The LED stripe is connected to the GPIO Pins 10 & 11.
The grabber is found, but I got the error "Device 'apa102' is disabled due to an error: 'Failed to open device (/dev/spidev0.0)", so the LED stay dark.
I found some threads which say to enable the SPI but there was no explanation on how-to...... With the old system everythings works just fine but who wants to miss out all the new, good stuff ;-) Could you maybe point me in the right direction?

**Reply by PaulePils** [reply]
Nevermind, I got i running.
I forgot to add "dtparam=spi=on" in the config.txt after creating a new SD card..... I feel so stupid....
